src/database/mod.rs

```diff
@@ -83,8 +83,44 @@ impl sqlx::Type<sqlx::Postgres> for PullRequestNumber {
     }
 }
 
+impl sqlx::Type<sqlx::Postgres> for TreeState {
+    fn type_info() -> sqlx::postgres::PgTypeInfo {
+        <i32 as sqlx::Type<sqlx::Postgres>>::type_info()
+    }
+}
+
+impl sqlx::Encode<'_, sqlx::Postgres> for TreeState {
+    fn encode_by_ref(
+        &self,
+        buf: &mut sqlx::postgres::PgArgumentBuffer,
+    ) -> Result<sqlx::encode::IsNull, Box<dyn std::error::Error + Send + Sync>> {
+        match self {
+            TreeState::Open => Ok(sqlx::encode::IsNull::Yes),
+            TreeState::Closed(priority) => {
+                <i32 as sqlx::Encode<sqlx::Postgres>>::encode(*priority as i32, buf)
+            }
+        }
+    }
+}
+
+impl sqlx::Decode<'_, sqlx::Postgres> for TreeState {
+    fn decode(value: sqlx::postgres::PgValueRef<'_>) -> Result<Self, sqlx::error::BoxDynError> {
+        let priority = <Option<i32> as sqlx::Decode<sqlx::Postgres>>::decode(value)?;
+        match priority {
+            None => Ok(TreeState::Open),
+            Some(p) => Ok(TreeState::Closed(p as u32)),
+        }
+    }
+}
+
+impl From<Option<TreeState>> for TreeState {
```

Comment:
This is a pretty weird impl, it's very unobvious if someone uses `.into()` to go from `Option<TreeState>` to `TreeState` and it magically selects this default through this impl. This should be set explicitly somewhere else.